packages/core/src/checkout-buttons/strategies/paypal-commerce/paypal-commerce-credit-button-strategy.ts

```diff
@@ -36,20 +40,35 @@ export default class PaypalCommerceCreditButtonStrategy implements CheckoutButto
             throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommercecredit" argument is not provided.`);
         }
 
-        const state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
-        const currencyCode = state.cart.getCartOrThrow().currency.code;
-        const paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
-        this._paypalCommerceSdk = await this._paypalScriptLoader.getPayPalSDK(paymentMethod, currencyCode, initializesOnCheckoutPage);
+        const { buyNowInitializeOptions, currencyCode, initializesOnCheckoutPage, messagingContainerId } = paypalcommercecredit;
 
-        this._renderButton(containerId, methodId, initializesOnCheckoutPage, style);
+        if (buyNowInitializeOptions) {
```

Comment:
One question do we need a whole different set of code block? The only difference I see is how currencyCode is passed? Maybe we can have something like

```
const currencyCode = currencyCode ||  cart.currency.code;

 if (!currencyCode) {
                throw new InvalidArgumentError(`Unable to initialize payment because "options.paypalcommercecredit.currencyCode" argument is not provided.`);
 }
 
 // rest          
```

Reply:
We should have another block of code due to the state variable. 
In buy now case we should not load default checkout, because we don't have access to checkout-settings, but for default flow we should

Reply:
You can check the difference on line 46 and 55